tests/pos/i12715/minimized.scala

```diff
@@ -0,0 +1,8 @@
+transparent inline def f: String =
+  inline 10 match
+    case _ =>
+      inline "foo" match
+        case x : String => x
```

Comment:
I'm wondering if we have the same issue with nested bindings such as in:
```scala
inline Some["foo"]("foo") match
        case Some(x) => x
```

Reply:
It does seem to break again in this case :/

Reply:
This one is related but not exactly the same issue. I opened #13161 to track this issue.